### PR TITLE
[BEAM-7286] RedisIO support for INCRBY/DECRBY operations

### PR DESCRIPTION
**Pull request by varundhussa**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by varundhussa** [comment]
@aromanenko-dev can you look at this?

File: sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java [hunk 1]
```diff
@@ -242,6 +242,62 @@ public void testWriteUsingHLLMethod() throws Exception {
     Assert.assertEquals(6, count);
   }
 
+  @Test
+  public void testWriteUsingINCR() throws Exception {
```
Comment (aromanenko-dev) [inline sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java]: Rename to `testWriteUsingINCRBY`

File: sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java [hunk 2]
```diff
@@ -242,6 +242,62 @@ public void testWriteUsingHLLMethod() throws Exception {
...
+        RedisIO.write().withEndpoint(REDIS_HOST, embeddedRedis.getPort()).withMethod(Method.INCR));
+
+    writePipeline.run();
+
+    long count = Long.parseLong(jedis.get(key));
+    Assert.assertEquals(11, count);
+  }
+
+  @Test
+  public void testWriteUsingDECR() throws Exception {
```
Comment (aromanenko-dev) [inline sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java]: Rename to `testWriteUsingDECRBY`

**Comment by aromanenko-dev** [comment]
Also, please rebase to avoid merge conflicts.

**Comment by varundhussa** [comment]
Thanks @aromanenko-dev. I've updated the PR. I didn't use the INCR/DECR operation because the class needs a KV object and the ops are key only. I can look at that if you think it is required in a different PR.

**Comment by aromanenko-dev** [comment]
LGTM
I think we can address INCR/DECR operation in other Jira/PR but before it would make sense to finish a discussion about RedisIO refactoring on ML

**Pull request merged**